### PR TITLE
release-21.1: assorted pgwire and sql shell backports

### PR DESCRIPTION
**Pull request by knz**
Backport:
  * 2/2 commits from "pgwire: only show TZ minutes offset if non-zero" (#65063)
  * 1/1 commits from "cli/sql: properly format common array types" (#66079)

Please see individual PRs for details.

/cc @cockroachdb/release

### CODE REVIEW

**Comment by rafiss** [comment]
i'll let @otan weigh in since he was opposed to the tz offset change being backported (https://github.com/cockroachdb/cockroach/pull/65341#pullrequestreview-661394014) -- but we can revisit this discussion

**Comment by otan** [comment]
Yeah I *think* it's too big a change for a minor version (if we think it fits, why not every version before as well?).

File: pkg/cli/sql_util.go [hunk 1]
```diff
@@ -1134,6 +1145,11 @@ func formatVal(
 		if colType == "FLOAT4" {
 			width = 32
 		}
+		if math.IsInf(t, 1) {
+			return "Infinity"
```
Comment (otan) [inline pkg/cli/sql_util.go:1149]: my concern is anyone relying on this right now on a major version who wants a quick patch to some other problem but this breaks something else unexpectedly.

**Comment by knz** [comment]
I'll let you two figure out whether to backport #65063.

I chose to backport it because otherwise my other PR #66079 did not apply cleanly. If we're unsure about the TZ changes, I can do a bit more work to backport #66079 without it. 

Just let me know what is best.

**Comment by knz** [comment]
would love an opinion on the above

**Review by otan — APPROVED** [review]
hmm, i guess the likelihood of someone having a really bad time with this is very slim.
since the rebase is effort too, and we're backporting it all the way down, i'm content with this being backported.

**Comment by knz** [comment]
ok cool, I'll send the backport PR for 20.2 too then.

Cheers

**Pull request merged**